### PR TITLE
Implement elective modules

### PR DESCRIPTION
**Pull request by GermanRanger**
Main features:
- elective modules are saved with new attribute realized with enum
- module can have one of three types: main module, placeholder module (to be shown in syllabus and course list) and elective module (to be only shown in syllabus)
- creating a new curriculum will show a new semester (unused 0) to assign elective modules (if unassigned elective modules are existent)
- added a new button and dropdown menu for creating new modules (or editing existent ones) to change the module type
- a click on elective modules in the course list will redirect to syllabus to the anchor where the elective module list begins
- the sequence of the courses in syllabus are reordered (elective modules are at the end)

### CODE REVIEW

File: src/main/java/rocks/turncodr/mycurriculum/model/ModuleType.java [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 
+/**
+ *  Attribute for Module.
+ */
 public enum ModuleType {
     MAIN_MODULE(0),
     PLACEHOLDER_MODULE(1),
     ELECTIVE_MODULE(2);
 
     private int index;
```
Comment (schmolli) [inline src/main/java/rocks/turncodr/mycurriculum/model/ModuleType.java:11]: Why do you need this index? Enums have an ordinal number starting at 0 for the first Object. Why do you not use that?

File: src/main/java/rocks/turncodr/mycurriculum/model/Module.java [hunk 2]
```diff
@@ -1,22 +1,23 @@
 package rocks.turncodr.mycurriculum.model;
 
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.Lob;
-import javax.persistence.ManyToOne;
+import javax.persistence.*;
```
Comment (schmolli) [inline src/main/java/rocks/turncodr/mycurriculum/model/Module.java:3]: imports with * are not a good practice.

File: src/main/java/rocks/turncodr/mycurriculum/application/lifecycle/DemoCurriculumLifecycleBean.java [hunk 3]
```diff
@@ -3,10 +3,7 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.SmartLifecycle;
 import org.springframework.stereotype.Component;
-import rocks.turncodr.mycurriculum.model.AreaOfStudies;
-import rocks.turncodr.mycurriculum.model.Curriculum;
-import rocks.turncodr.mycurriculum.model.ExReg;
-import rocks.turncodr.mycurriculum.model.Module;
+import rocks.turncodr.mycurriculum.model.*;
```
Comment (schmolli) [inline src/main/java/rocks/turncodr/mycurriculum/application/lifecycle/DemoCurriculumLifecycleBean.java:6]: imports with * are not a good practice.

**Comment by GermanRanger** [comment]
@schmolli Code adjusted according to your mentions.

**Pull request merged**